### PR TITLE
[Model] mPLUG-Owl3 Added

### PR DESCRIPTION
**Pull request by SYuan03**
Link: https://huggingface.co/mPLUG/mPLUG-Owl3-7B-240728

1. On RealWorldQA, achieve Overall  0.48627450980392156
image
2. add support in tools [version map]

More:
1. The chat_inner interface has not been completed.

### CODE REVIEW

File: vlmeval/vlm/mplug_owl3.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    INTERLEAVE = True
+    INSTALL_REQ_TXT = 'https://github.com/X-PLUG/mPLUG-Owl/blob/main/mPLUG-Owl3/requirements.txt'
+
+    def __init__(self, model_path=None, **kwargs):
+        assert model_path is not None
+        self.tokenizer = AutoTokenizer.from_pretrained(
+            model_path,
+            trust_remote_code=True
+        )
+        from mplug_owl3_dsy import mPLUGOwl3Model
```
Comment (kennymckormick) [inline vlmeval/vlm/mplug_owl3.py:21]: Do not include custom names in the PR.
You can refer to the implementation of mPLUG-Owl2 and assume that mPLUG-Owl3 has been installed properly.

File: vlmeval/vlm/mplug_owl3.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+                prompt_full += '<|image|>'
+            elif msg['type'] == 'text':
+                prompt_full += msg['value']
+
+        needed_messages = [
+            {'role': 'user', 'content': prompt_full},
+            {'role': 'assistant', 'content': ''}
+        ]
+
+        self.logger.info(f'needed_messages: {needed_messages}')
```
Comment (kennymckormick) [inline vlmeval/vlm/mplug_owl3.py:118]: Remove those loggings

File: vlmeval/vlm/mplug_owl3.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+        inputs.to('cuda')
+        inputs.update({
+            'tokenizer': self.tokenizer,
+            'max_new_tokens': 1024,
+            'decode_text': True,
+        })
+
+        g = self.model.generate(**inputs)
+        print(g)
```
Comment (kennymckormick) [inline vlmeval/vlm/mplug_owl3.py:134]: remove the print

**Comment by kennymckormick** [comment]
@SYuan03 Please resolve the comments and fix the conflicts.

File: vlmeval/vlm/mplug_owl3.py (hunk 1 above)
Comment (SYuan03) [reply vlmeval/vlm/mplug_owl3.py:21]: Sorry, here i was trying to test the difference between loading with automodel and the way it's loaded in the official demo, I'll fix this.

**Review by kennymckormick — APPROVED** [review]

**Pull request merged**